autoscaler/main.go

```diff
@@ -262,3 +274,23 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func isMetricsServerInstalled(mgr ctrl.Manager, logger logr.Logger) bool {
+	var metricsServerDeployment appsv1.Deployment
+
+	// Use APIReader (uncached client) for direct access to the API server
+	// uses because mgr not cache the metrics-server deployment
+	err := mgr.GetAPIReader().Get(context.TODO(), types.NamespacedName{Name: "metrics-server", Namespace: "kube-system"}, &metricsServerDeployment)
+	if err != nil {
+		if client.IgnoreNotFound(err) != nil {
```

Comment:
nit: the `IgnoreNotFound` is for when you want to return this error.

You can use `apierrors.IsNotFound(err)` which is more appropriate in this case